scripts/ci/pr-bot/shared/userCommand.ts

```diff
@@ -0,0 +1,231 @@
+    await github.nextActionReviewers(pullNumber, existingLabels);
+    prState.nextAction = "Reviewers";
+
+    // Persist state
+    await stateClient.writePrState(pullNumber, prState);
+    await stateClient.writeReviewersForLabelState(
+      labelOfReviewer,
+      reviewersState
+    );
+  }
+}
+
+// If they've manually assigned a reviewer, just silence notifications and ignore this pr going forward.
+// TODO(damccorm) - we could try to do something more intelligent here like figuring out which label that reviewer belongs to.
+async function manuallyAssignedToReviewer(
+  pullNumber: number,
+  stateClient: typeof StateClient
+) {
+  await stopReviewerNotifications(
+    pullNumber,
+    stateClient,
+    "review requested by someone other than the bot, ceding control"
+  );
+}
+
+async function stopReviewerNotifications(
+  pullNumber: number,
+  stateClient: typeof StateClient,
+  reason: string
+) {
+  let prState = await stateClient.getPrState(pullNumber);
+  prState.stopReviewerNotifications = true;
+  await stateClient.writePrState(pullNumber, prState);
+
+  // Comment acknowledging command
+  await github.addPrComment(
+    pullNumber,
+    commentStrings.stopNotifications(reason)
+  );
+}
+
+async function remindAfterTestsPass(
+  pullNumber: number,
+  username: string,
+  stateClient: typeof StateClient
+) {
+  let prState = await stateClient.getPrState(pullNumber);
+  prState.remindAfterTestsPass.push(username);
+  await stateClient.writePrState(pullNumber, prState);
+
+  // Comment acknowledging command
+  await github.addPrComment(
+    pullNumber,
+    commentStrings.remindReviewerAfterTestsPass(username)
+  );
+}
+
+async function waitOnAuthor(
+  payload: any,
+  pullNumber: number,
+  stateClient: typeof StateClient
+) {
+  const existingLabels = payload.issue?.labels || payload.pull_request?.labels;
+  await github.nextActionAuthor(pullNumber, existingLabels);
+  let prState = await stateClient.getPrState(pullNumber);
+  prState.nextAction = "Author";
+  await stateClient.writePrState(pullNumber, prState);
+}
+
+async function assignReviewerSet(
+  payload: any,
+  pullNumber: number,
+  stateClient: typeof StateClient,
+  reviewerConfig: typeof ReviewerConfig
+) {
+  let prState = await stateClient.getPrState(pullNumber);
+  if (Object.values(prState.reviewersAssignedForLabels).length > 0) {
+    await github.addPrComment(
+      pullNumber,
+      commentStrings.reviewersAlreadyAssigned(
+        Object.values(prState.reviewersAssignedForLabels)
+      )
+    );
+    return;
+  }
+
+  const existingLabels = payload.issue?.labels || payload.pull_request?.labels;
+  const pullAuthor =
+    payload.issue?.user?.login || payload.pull_request?.user?.login;
+  const reviewersForLabels = reviewerConfig.getReviewersForLabels(
+    existingLabels,
+    [pullAuthor]
+  );
+  let reviewerStateToUpdate = {};
+  var labels = Object.keys(reviewersForLabels);
+  if (!labels || labels.length == 0) {
```

Comment:
`labels?.length > 0`

Reply:
That gets pretty unclear when labels is null, especially because technically the whole condition would be `!(labels?.length > 0)`. For readability, I think I prefer the explicit `||`